src/main/java/seedu/address/model/person/GithubUsername.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+
+/**
+ * Represents a Person's GitHub Username in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGithubUsername(String)}
+ */
+public class GithubUsername {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Github usernames should only contain at most 39 alphanumeric characters and hyphens,"
+                    + " but cannot have consecutive hyphens and cannot begin or end with hyphens.";
+
+    /*
+     * The first and last character of the GitHub Username must a hyphen,
+     * can't have consecutive hyphens in the GitHub Username,
+     * and can't be longer than 39 alphanumeric characters.
```

Comment:
* The first and last character of the GitHub Username must not be a hyphen,
* can't have consecutive hyphens in the GitHub Username,
* and can't be longer than 39 alphanumeric characters.

Reply:
Thanks for the catch. I will update it.